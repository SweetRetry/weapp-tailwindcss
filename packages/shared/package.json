{
  "name": "@weapp-tailwindcss/shared",
  "version": "0.0.0",
  "description": "@weapp-tailwindcss/shared",
  "author": "ice breaker <1324318532@qq.com>",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/sonofmagic/monorepo-template.git",
    "directory": "packages/bar"
  },
  "bugs": {
    "url": "https://github.com/sonofmagic/monorepo-template/issues"
  },
  "keywords": [],
  "sideEffects": false,
  "exports": {
    ".": "./src/index.ts"
  },
  "files": [
    "dist"
  ],
  "scripts": {
    "dev": "tsup --watch --sourcemap",
    "build": "tsup",
    "test": "vitest run",
    "test:dev": "vitest",
    "release": "pnpm publish",
    "lint": "eslint .",
    "lint:fix": "eslint . --fix"
  },
  "publishConfig": {
    "exports": {
      ".": {
        "types": "./dist/index.d.ts",
        "import": "./dist/index.mjs",
        "require": "./dist/index.js"
      }
    },
    "main": "./dist/index.js",
    "module": "./dist/index.mjs",
    "types": "./dist/index.d.ts"
  },
  "dependencies": {}
}
