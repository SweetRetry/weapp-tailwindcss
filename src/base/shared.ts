import groupBy from 'lodash.groupby'
import type { UserDefinedOptions } from '@/types'

export function getGroupedEntries<T> (entries: [string, T][], options: Required<UserDefinedOptions>) {
  const { cssMatcher, htmlMatcher, jsMatcher } = options
  const groupedEntries = groupBy(entries, ([file]) => {
    if (cssMatcher(file)) {
      return 'css'
    } else if (htmlMatcher(file)) {
      return 'html'
    } else if (jsMatcher(file)) {
      return 'js'
    } else {
      return 'other'
    }
  })
  return groupedEntries
}
