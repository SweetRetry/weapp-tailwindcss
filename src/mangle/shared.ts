export const mangleClassPrefix = 'MANGLE__'

export const mangleClassSuffix = '__MANGLE'

export const mangleClassRegex = /MANGLE__[a-zA-Z0-9_-]+__MANGLE/g

export function markForMangled (str: string) {
  if (typeof str === 'string' && str) {
    return `${mangleClassPrefix}${str.trim()}${mangleClassSuffix}`
  }
  return str
}
