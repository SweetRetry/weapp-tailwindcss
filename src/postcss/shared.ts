import { escape } from '@/escape'
import { SimpleMappingChars2String } from '@/dic'
// css 中，要多加一个 '\' 来转义
// for raw css selector
// export function cssSelectorReplacer(selector: string, escapeEntries = MappingChars2StringEntries) {
//   return escape(selector, true, escapeEntries).replace(/\\2c /g, dic[','])
// }

export function internalCssSelectorReplacer(selectors: string, map: Record<string, string> = SimpleMappingChars2String) {
  return escape(selectors, {
    map
  })
}
