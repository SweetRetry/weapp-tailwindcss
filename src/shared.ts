import path from 'path'

export { styleHandler } from './postcss'
export { templeteHandler } from './wxml'
export { jsxHandler } from './jsx'
export const pluginName = 'weapp-tailwindcss-webpack-plugin'

export function getFileName (file: string) {
  return path.basename(file, path.extname(file))
}
