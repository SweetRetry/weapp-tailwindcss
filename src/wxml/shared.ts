import { MappingChars2String as dic } from '@/dic'
import { ICommonReplaceOptions } from '@/types'
import { mangleMark } from '@/mangle/expose'

// export function arbitraryValuesReplacer (str: string) {}

export function replaceWxml (
  original: string,
  options: ICommonReplaceOptions | boolean = {
    keepEOL: false,
    mangle: false
  }
) {
  if (typeof options === 'boolean') {
    options = {
      keepEOL: options,
      mangle: false
    }
  }
  let res = original
    .replace(/\[/g, dic['['])
    .replace(/\]/g, dic[']'])
    .replace(/\(/g, dic['('])
    .replace(/\)/g, dic[')'])
    .replace(/#/g, dic['#']) // hex
    .replace(/!/g, dic['!']) // css !important
    .replace(/\//g, dic['/'])
    .replace(/\./g, dic['.'])
    .replace(/:/g, dic[':'])
    // https://github.com/sonofmagic/weapp-tailwindcss-webpack-plugin/issues/8
    .replace(/%/g, dic['%'])
    .replace(/,/g, dic[','])
    .replace(/\\/g, dic['\\'])
    .replace(/'/g, dic["'"])
    .replace(/"/g, dic['"'])
    .replace(/\*/g, dic['*'])
    .replace(/&/g, dic['&'])
    .replace(/@/g, dic['@'])
    .replace(/{/g, dic['{'])
    .replace(/}/g, dic['}'])
  if (!options.keepEOL) {
    res = res
      // 去除无用换行符和空格
      .replace(/[\r\n]+/g, '')
  }
  if (options.mangle) {
    res = mangleMark(res, options.classGenerator)
  }
  return res
}
