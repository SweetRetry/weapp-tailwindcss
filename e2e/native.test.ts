import fs from 'node:fs/promises'
import path from 'node:path'
import automator from 'miniprogram-automator'
import { deleteAsync } from 'del'
import type { ProjectEntry } from './shared'
import { formatWxml, loadCss, projectFilter, removeWxmlId, twExtract, wait } from './shared'

const TestProjectsEntries: ProjectEntry[] = [
  {
    name: 'native',
    projectPath: 'native',
    testMethod: async (_, projectPath) => {
      expect(await loadCss(path.resolve(projectPath, 'dist/app.wxss'))).toMatchSnapshot('css')
    },
  },

  {
    name: 'native-ts',
    projectPath: 'native-ts',
    testMethod: async (_, projectPath) => {
      expect(await loadCss(path.resolve(projectPath, 'dist/app.scss'))).toMatchSnapshot('css')
    },
  },
  {
    name: 'web-postcss7-compat',
    projectPath: 'web-postcss7-compat',
    skipOpenAutomator: true,
    testMethod: async (_, projectPath) => {
      expect(await loadCss(path.resolve(projectPath, 'result.css'))).toMatchSnapshot('css')
    },
  },
  // {
  //   name: 'native-skyline',
  //   projectPath: 'native-skyline',
  //   testMethod: async (_, projectPath) => {
  //     expect(await loadCss(path.resolve(projectPath, 'dist/app.wxss'))).toMatchSnapshot('css')
  //   },
  // },
  // {
  //   name: 'native-ts-skyline',
  //   projectPath: 'native-ts-skyline',
  //   testMethod: async (_, projectPath) => {
  //     expect(await loadCss(path.resolve(projectPath, 'dist/app.wxss'))).toMatchSnapshot('css')
  //   },
  // },
]

describe('e2e native', () => {
  it.each(projectFilter(TestProjectsEntries))('$name', async (config) => {
    const projectPath = path.resolve(__dirname, '../apps', config.projectPath)
    const testMethod = config.testMethod
    const root = path.resolve(__dirname, '../apps', config.name)
    await deleteAsync([path.resolve(root, 'node_modules/.cache')])
    await twExtract(root)
    const json = await fs.readFile(path.resolve(root, '.tw-patch/tw-class-list.json'), 'utf8')
    expect(json).toMatchSnapshot('json')

    if (config.skipOpenAutomator) {
      await testMethod(null, projectPath)
      return
    }
    const miniProgram = await automator.launch({
      // cliPath: 'C:\\Program Files (x86)\\Tencent\\微信web开发者工具\\cli.bat',
      projectPath,
    })
    const page = await miniProgram.reLaunch(config.url ?? '/pages/index/index')

    if (page) {
      await testMethod(page, projectPath)
      const pageEl = await page.$('page')
      let wxml = await pageEl?.wxml()
      if (wxml) {
        wxml = removeWxmlId(wxml)
        try {
          wxml = await formatWxml(wxml)
        }
        catch {
          console.error(`parse error: ${config.projectPath}`)
        }

        expect(wxml).toMatchSnapshot('wxml')
      }

      await page.waitFor(3000)
    }

    await miniProgram.close()
    await wait()
  })
})
