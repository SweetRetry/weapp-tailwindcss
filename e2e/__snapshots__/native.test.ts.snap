// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`e2e native > 'vite-native' > css 1`] = `
".hahaha {
  color: #bc8f8f;
}
/*$vite$:1*/
"
`;

exports[`e2e native > 'vite-native' > json 1`] = `
"[
  "bg-[#7d7ac2]",
  "block",
  "h-[30px]",
  "h-[45px]",
  "text-[#123456]",
  "text-[100px]",
  "w-[323px]"
]
"
`;

exports[`e2e native > 'vite-native' > wxml 1`] = `
"<navigation-bar is="components/navigation-bar/navigation-bar">
  <view class="weui-navigation-bar bar--weui-navigation-bar">
    <view
      class="bg-_h232323_ bar--bg-_h232323_ weui-navigation-bar__inner bar--weui-navigation-bar__inner ios bar--ios"
      style="color:; background:; padding-right: 94px; height: calc(var(--height) + 47px); padding-top: 47px">
      1111
      <view class="weui-navigation-bar__left bar--weui-navigation-bar__left" style="width: 94px">
        <view class="weui-navigation-bar__buttons bar--weui-navigation-bar__buttons weui-navigation-bar__buttons_goback bar--weui-navigation-bar__buttons_goback">
          <view
            aria-label="返回"
            aria-role="button"
            role="button"
            class="weui-navigation-bar__btn_goback_wrapper bar--weui-navigation-bar__btn_goback_wrapper"
            hover-class="weui-active"
            hover-stay-time="100">
            <view class="weui-navigation-bar__button bar--weui-navigation-bar__button weui-navigation-bar__btn_goback bar--weui-navigation-bar__btn_goback"></view>
          </view>
        </view>
      </view>
      <view class="weui-navigation-bar__center bar--weui-navigation-bar__center"></view>
      <view class="weui-navigation-bar__right bar--weui-navigation-bar__right"></view>
    </view>
  </view>
</navigation-bar>
<scroll-view scroll-y="" type="list">
  { "BASE_URL": "/", "DEV": false, "MODE": "production", "MP_PLATFORM": "weapp", "PROD": true, "SSR": false, "VITE_ENV": ".env"}
  <view class="bg-_h7d7ac2_ text-_100px_ text-_h123456_ w-_323px_ h- 30px">111</view>
  <view>Hello MINA!</view>
  <view>1</view>
  <view>2</view>
  <view>3</view>
  <view>4</view>
  <view>5</view>
  <view>FirstName: Hulk, LastName: Hu</view>
  <view>FirstName: Shang, LastName: You</view>
  <view>FirstName: Gideon, LastName: Lin</view>
  <view>hello world</view>
  <view data-hi="Weixin">Click me!</view>
</scroll-view>
<view>odd</view>
<view>even</view>
<view>odd</view>
<view>even</view>
<view>odd</view>
.luna-dom-highlighter{position:fixed;left:0;top:0;width:100%;height:100%;z-index:100000;pointer-events:none;font-size:13px}.luna-dom-highlighter-fill{position:absolute;top:0;right:0;bottom:0;left:0}.luna-dom-highlighter-platform-linux{font-family:Roboto,Ubuntu,Arial,sans-serif}.luna-dom-highlighter-platform-mac{color:#303942;font-family:'.SFNSDisplay-Regular','Helvetica
Neue','Lucida Grande',sans-serif}.luna-dom-highlighter-platform-windows{font-family:'Segoe
UI',Tahoma,sans-serif}.luna-dom-highlighter-px{color:gray}#luna-dom-highlighter-element-title{position:absolute;z-index:10}.luna-dom-highlighter-tooltip-content{position:absolute;-webkit-user-select:none;-moz-user-select:none;user-select:none;background-color:#fff;padding:5px
8px;border:1px solid #fff;border-radius:3px;box-sizing:border-box;min-width:100px;max-width:min(300px,100% -
4px);z-index:2;background-clip:padding-box;will-change:transform;text-rendering:optimizeLegibility;pointer-events:none;filter:drop-shadow(0 2px 4px
rgba(0,0,0,.35))}.luna-dom-highlighter-tooltip-content
.luna-dom-highlighter-tooltip-arrow{background:#fff;width:15px;height:8px;position:absolute}.luna-dom-highlighter-element-info-section{margin-top:12px;margin-bottom:6px}.luna-dom-highlighter-section-name{color:#333;font-weight:500;font-size:10px;text-transform:uppercase;letter-spacing:.05em;line-height:12px}.luna-dom-highlighter-element-info{display:flex;flex-direction:column}.luna-dom-highlighter-element-info-header{display:flex;align-items:center}.luna-dom-highlighter-element-info-body{display:flex;flex-direction:column;padding-top:2px;margin-top:2px}.luna-dom-highlighter-element-info-row{display:flex;line-height:19px}.luna-dom-highlighter-separator-container{display:flex;align-items:center;flex:auto;margin-left:7px}.luna-dom-highlighter-separator{border-top:1px
solid
#ddd;width:100%}.luna-dom-highlighter-element-info-name{flex-shrink:0;color:#666}.luna-dom-highlighter-element-info-gap{flex:auto}.luna-dom-highlighter-element-info-value-color{display:flex;color:#303942;margin-left:10px;align-items:baseline}.luna-dom-highlighter-a11y-icon{width:16px;height:16px;background-repeat:no-repeat;display:inline-block}.luna-dom-highlighter-element-info-value-contrast{display:flex;align-items:center;text-align:right;color:#303942;margin-left:10px}.luna-dom-highlighter-element-info-value-contrast
.luna-dom-highlighter-a11y-icon{margin-left:8px}.luna-dom-highlighter-element-info-value-icon{display:flex;align-items:center}.luna-dom-highlighter-element-info-value-text{text-align:right;color:#303942;margin-left:10px;align-items:baseline;overflow:hidden;text-overflow:ellipsis;white-space:nowrap}.luna-dom-highlighter-color-swatch{display:flex;margin-right:2px;width:10px;height:10px;background-image:url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAwAAAAMCAIAAADZF8uwAAAAGUlEQVQYV2M4gwH+YwCGIasIUwhT25BVBADtzYNYrHvv4gAAAABJRU5ErkJggg==);line-height:10px}.luna-dom-highlighter-color-swatch-inner{flex:auto;border:1px
solid
#808002}.luna-dom-highlighter-element-layout-type{margin-right:10px;width:16px;height:16px}.luna-dom-highlighter-element-layout-type.luna-dom-highlighter-grid{background-image:url('data:image/svg+xml,')}.luna-dom-highlighter-element-layout-type.luna-dom-highlighter-flex{background-image:url('data:image/svg+xml,')}.luna-dom-highlighter-element-description{flex:1
1;font-weight:700;word-wrap:break-word;word-break:break-all}.luna-dom-highlighter-dimensions{color:#737373;text-align:right;margin-left:10px}.luna-dom-highlighter-material-node-width{margin-right:2px}.luna-dom-highlighter-material-node-height{margin-left:2px}.luna-dom-highlighter-material-tag-name{color:#881280}.luna-dom-highlighter-material-class-name,.luna-dom-highlighter-material-node-id{color:#1a1aa6}.luna-dom-highlighter-contrast-text{width:16px;height:16px;text-align:center;line-height:16px;margin-right:8px;border:1px
solid #000;padding:0
1px}.luna-dom-highlighter-a11y-icon-not-ok{background-image:url('data:image/svg+xml,')}.luna-dom-highlighter-a11y-icon-warning{background-image:url('data:image/svg+xml,')}.luna-dom-highlighter-a11y-icon-ok{background-image:url('data:image/svg+xml,')}@media
(forced-colors:active){:root,body{background-color:transparent;forced-color-adjust:none}.luna-dom-highlighter-tooltip-content{border-color:Highlight;background-color:canvas;color:text;forced-color-adjust:none}.luna-dom-highlighter-tooltip-content::after{background-color:Highlight}.luna-dom-highlighter-color-swatch-inner,.luna-dom-highlighter-contrast-text,.luna-dom-highlighter-separator{border-color:Highlight}.luna-dom-highlighter-section-name{color:Highlight}.luna-dom-highlighter-dimensions,.luna-dom-highlighter-element-info-name,.luna-dom-highlighter-element-info-value-color,.luna-dom-highlighter-element-info-value-contrast,.luna-dom-highlighter-element-info-value-icon,.luna-dom-highlighter-element-info-value-text,.luna-dom-highlighter-material-class-name,.luna-dom-highlighter-material-node-id,.luna-dom-highlighter-material-tag-name{color:canvastext}}.luna-dom-highlighter{z-index:2147483647;}
"
`;

exports[`e2e native > 'vite-native-ts' > css 1`] = `
"::before,
::after {
  --tw-content: '';
}
view,
text,
:before,
:after {
  --tw-border-spacing-x: 0;
  --tw-border-spacing-y: 0;
  --tw-translate-x: 0;
  --tw-translate-y: 0;
  --tw-rotate: 0;
  --tw-skew-x: 0;
  --tw-skew-y: 0;
  --tw-scale-x: 1;
  --tw-scale-y: 1;
  --tw-pan-x: ;
  --tw-pan-y: ;
  --tw-pinch-zoom: ;
  --tw-scroll-snap-strictness: proximity;
  --tw-gradient-from-position: ;
  --tw-gradient-via-position: ;
  --tw-gradient-to-position: ;
  --tw-ordinal: ;
  --tw-slashed-zero: ;
  --tw-numeric-figure: ;
  --tw-numeric-spacing: ;
  --tw-numeric-fraction: ;
  --tw-ring-inset: ;
  --tw-ring-offset-width: 0px;
  --tw-ring-offset-color: #fff;
  --tw-ring-color: rgb(59 130 246 / 0.5);
  --tw-ring-offset-shadow: 0 0 #0000;
  --tw-ring-shadow: 0 0 #0000;
  --tw-shadow: 0 0 #0000;
  --tw-shadow-colored: 0 0 #0000;
  --tw-blur: ;
  --tw-brightness: ;
  --tw-contrast: ;
  --tw-grayscale: ;
  --tw-hue-rotate: ;
  --tw-invert: ;
  --tw-saturate: ;
  --tw-sepia: ;
  --tw-drop-shadow: ;
  --tw-backdrop-blur: ;
  --tw-backdrop-brightness: ;
  --tw-backdrop-contrast: ;
  --tw-backdrop-grayscale: ;
  --tw-backdrop-hue-rotate: ;
  --tw-backdrop-invert: ;
  --tw-backdrop-opacity: ;
  --tw-backdrop-saturate: ;
  --tw-backdrop-sepia: ;
  --tw-contain-size: ;
  --tw-contain-layout: ;
  --tw-contain-paint: ;
  --tw-contain-style: ;
  box-sizing: border-box;
  border-width: 0;
  border-style: solid;
  border-color: currentColor;
}
::backdrop {
  --tw-border-spacing-x: 0;
  --tw-border-spacing-y: 0;
  --tw-translate-x: 0;
  --tw-translate-y: 0;
  --tw-rotate: 0;
  --tw-skew-x: 0;
  --tw-skew-y: 0;
  --tw-scale-x: 1;
  --tw-scale-y: 1;
  --tw-pan-x: ;
  --tw-pan-y: ;
  --tw-pinch-zoom: ;
  --tw-scroll-snap-strictness: proximity;
  --tw-gradient-from-position: ;
  --tw-gradient-via-position: ;
  --tw-gradient-to-position: ;
  --tw-ordinal: ;
  --tw-slashed-zero: ;
  --tw-numeric-figure: ;
  --tw-numeric-spacing: ;
  --tw-numeric-fraction: ;
  --tw-ring-inset: ;
  --tw-ring-offset-width: 0px;
  --tw-ring-offset-color: #fff;
  --tw-ring-color: rgb(59 130 246 / 0.5);
  --tw-ring-offset-shadow: 0 0 #0000;
  --tw-ring-shadow: 0 0 #0000;
  --tw-shadow: 0 0 #0000;
  --tw-shadow-colored: 0 0 #0000;
  --tw-blur: ;
  --tw-brightness: ;
  --tw-contrast: ;
  --tw-grayscale: ;
  --tw-hue-rotate: ;
  --tw-invert: ;
  --tw-saturate: ;
  --tw-sepia: ;
  --tw-drop-shadow: ;
  --tw-backdrop-blur: ;
  --tw-backdrop-brightness: ;
  --tw-backdrop-contrast: ;
  --tw-backdrop-grayscale: ;
  --tw-backdrop-hue-rotate: ;
  --tw-backdrop-invert: ;
  --tw-backdrop-opacity: ;
  --tw-backdrop-saturate: ;
  --tw-backdrop-sepia: ;
  --tw-contain-size: ;
  --tw-contain-layout: ;
  --tw-contain-paint: ;
  --tw-contain-style: ;
}
.container {
  width: 100%;
}
@media (min-width: 640px) {
  .container {
    max-width: 640px;
  }
}
@media (min-width: 768px) {
  .container {
    max-width: 768px;
  }
}
@media (min-width: 1024px) {
  .container {
    max-width: 1024px;
  }
}
@media (min-width: 1280px) {
  .container {
    max-width: 1280px;
  }
}
@media (min-width: 1536px) {
  .container {
    max-width: 1536px;
  }
}
.block {
  display: block;
}
.bg-_h323284_ {
  --tw-bg-opacity: 1;
  background-color: rgb(50 50 132 / var(--tw-bg-opacity, 1));
}
.text-_h31c034_ {
  --tw-text-opacity: 1;
  color: rgb(49 192 52 / var(--tw-text-opacity, 1));
}
/*$vite$:1*/
"
`;

exports[`e2e native > 'vite-native-ts' > json 1`] = `
"[
  "bg-[#323284]",
  "block",
  "container",
  "text-[#31c034]"
]
"
`;

exports[`e2e native > 'vite-native-ts' > wxml 1`] = `"<scroll-view scroll-y="" class="scrollarea" type="list" enable-back-to-top=""><view class="container"><view class="userinfo"><button class="avatar-wrapper" open-type="chooseAvatar" role="button" aria-disabled="false"><image class="avatar" src="https://mmbiz.qpic.cn/mmbiz/icTdbqWNOwNRna42FI242Lcia07jQodd2FJGIYQfG0LAJGFxM4FbnQP6yfMxBgJ0F3YRqJCJ1aPAK2dQagdusBZg/0" role="img"></image></button><view class="nickname-wrapper"><text class="nickname-label"></text><input class="nickname-input" placeholder="请输入昵称" type="nickname">请输入昵称</input></view></view><view class="usermotto"><text class="user-motto text-_h31c034_ bg-_h323284_"></text></view></view></scroll-view>.luna-dom-highlighter{position:fixed;left:0;top:0;width:100%;height:100%;z-index:100000;pointer-events:none;font-size:13px}.luna-dom-highlighter-fill{position:absolute;top:0;right:0;bottom:0;left:0}.luna-dom-highlighter-platform-linux{font-family:Roboto,Ubuntu,Arial,sans-serif}.luna-dom-highlighter-platform-mac{color:#303942;font-family:'.SFNSDisplay-Regular','Helvetica Neue','Lucida Grande',sans-serif}.luna-dom-highlighter-platform-windows{font-family:'Segoe UI',Tahoma,sans-serif}.luna-dom-highlighter-px{color:gray}#luna-dom-highlighter-element-title{position:absolute;z-index:10}.luna-dom-highlighter-tooltip-content{position:absolute;-webkit-user-select:none;-moz-user-select:none;user-select:none;background-color:#fff;padding:5px 8px;border:1px solid #fff;border-radius:3px;box-sizing:border-box;min-width:100px;max-width:min(300px,100% - 4px);z-index:2;background-clip:padding-box;will-change:transform;text-rendering:optimizeLegibility;pointer-events:none;filter:drop-shadow(0 2px 4px rgba(0,0,0,.35))}.luna-dom-highlighter-tooltip-content .luna-dom-highlighter-tooltip-arrow{background:#fff;width:15px;height:8px;position:absolute}.luna-dom-highlighter-element-info-section{margin-top:12px;margin-bottom:6px}.luna-dom-highlighter-section-name{color:#333;font-weight:500;font-size:10px;text-transform:uppercase;letter-spacing:.05em;line-height:12px}.luna-dom-highlighter-element-info{display:flex;flex-direction:column}.luna-dom-highlighter-element-info-header{display:flex;align-items:center}.luna-dom-highlighter-element-info-body{display:flex;flex-direction:column;padding-top:2px;margin-top:2px}.luna-dom-highlighter-element-info-row{display:flex;line-height:19px}.luna-dom-highlighter-separator-container{display:flex;align-items:center;flex:auto;margin-left:7px}.luna-dom-highlighter-separator{border-top:1px solid #ddd;width:100%}.luna-dom-highlighter-element-info-name{flex-shrink:0;color:#666}.luna-dom-highlighter-element-info-gap{flex:auto}.luna-dom-highlighter-element-info-value-color{display:flex;color:#303942;margin-left:10px;align-items:baseline}.luna-dom-highlighter-a11y-icon{width:16px;height:16px;background-repeat:no-repeat;display:inline-block}.luna-dom-highlighter-element-info-value-contrast{display:flex;align-items:center;text-align:right;color:#303942;margin-left:10px}.luna-dom-highlighter-element-info-value-contrast .luna-dom-highlighter-a11y-icon{margin-left:8px}.luna-dom-highlighter-element-info-value-icon{display:flex;align-items:center}.luna-dom-highlighter-element-info-value-text{text-align:right;color:#303942;margin-left:10px;align-items:baseline;overflow:hidden;text-overflow:ellipsis;white-space:nowrap}.luna-dom-highlighter-color-swatch{display:flex;margin-right:2px;width:10px;height:10px;background-image:url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAwAAAAMCAIAAADZF8uwAAAAGUlEQVQYV2M4gwH+YwCGIasIUwhT25BVBADtzYNYrHvv4gAAAABJRU5ErkJggg==);line-height:10px}.luna-dom-highlighter-color-swatch-inner{flex:auto;border:1px solid #808002}.luna-dom-highlighter-element-layout-type{margin-right:10px;width:16px;height:16px}.luna-dom-highlighter-element-layout-type.luna-dom-highlighter-grid{background-image:url('data:image/svg+xml,')}.luna-dom-highlighter-element-layout-type.luna-dom-highlighter-flex{background-image:url('data:image/svg+xml,')}.luna-dom-highlighter-element-description{flex:1 1;font-weight:700;word-wrap:break-word;word-break:break-all}.luna-dom-highlighter-dimensions{color:#737373;text-align:right;margin-left:10px}.luna-dom-highlighter-material-node-width{margin-right:2px}.luna-dom-highlighter-material-node-height{margin-left:2px}.luna-dom-highlighter-material-tag-name{color:#881280}.luna-dom-highlighter-material-class-name,.luna-dom-highlighter-material-node-id{color:#1a1aa6}.luna-dom-highlighter-contrast-text{width:16px;height:16px;text-align:center;line-height:16px;margin-right:8px;border:1px solid #000;padding:0 1px}.luna-dom-highlighter-a11y-icon-not-ok{background-image:url('data:image/svg+xml,')}.luna-dom-highlighter-a11y-icon-warning{background-image:url('data:image/svg+xml,')}.luna-dom-highlighter-a11y-icon-ok{background-image:url('data:image/svg+xml,')}@media (forced-colors:active){:root,body{background-color:transparent;forced-color-adjust:none}.luna-dom-highlighter-tooltip-content{border-color:Highlight;background-color:canvas;color:text;forced-color-adjust:none}.luna-dom-highlighter-tooltip-content::after{background-color:Highlight}.luna-dom-highlighter-color-swatch-inner,.luna-dom-highlighter-contrast-text,.luna-dom-highlighter-separator{border-color:Highlight}.luna-dom-highlighter-section-name{color:Highlight}.luna-dom-highlighter-dimensions,.luna-dom-highlighter-element-info-name,.luna-dom-highlighter-element-info-value-color,.luna-dom-highlighter-element-info-value-contrast,.luna-dom-highlighter-element-info-value-icon,.luna-dom-highlighter-element-info-value-text,.luna-dom-highlighter-material-class-name,.luna-dom-highlighter-material-node-id,.luna-dom-highlighter-material-tag-name{color:canvastext}}.luna-dom-highlighter{z-index:2147483647;}"`;

exports[`e2e native > 'web-postcss7-compat' > css 1`] = `
".h-\\[121px\\] {
  height: 121px;
}
.w-\\[99px\\] {
  width: 99px;
}
"
`;

exports[`e2e native > 'web-postcss7-compat' > json 1`] = `
"[]
"
`;
